src/ui/atoms/progressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react'
+import type { DeepReadonly } from 'superTypes'
+import { Typography } from '../typography/Typography'
+import './progressBar.scss'
+import classNames from 'classnames'
+import { toPercent } from 'utils'
+
+export type TProgressBarProps = {
+  /**
+   * The progress bar title.
+   */
+  readonly label?: string
+  /**
+   * The start value of the progress bar.
+   */
+  readonly minValue?: number
+  /**
+   * The end value of the progress bar.
+   */
+  readonly maxValue?: number
+  /**
+   * The current value of the progress bar.
+   */
+  readonly currentValue?: number
+  /**
+   * Allows to define how to display the current value.
+   *
+   * @param currentValue The current value of the progress bar.
+   * @return Returns the formatted current value.
+   */
+  readonly currentValueFormatter?: (currentValue: number) => string
+  /**
+   * Allows to define how to display the progress
+   *
+   * @param currentValue The current value of the progress bar.
+   * @param minValue The start value of the progress bar.
+   * @param maxValue The end value of the progress bar.
+   * @return Returns the formatted progress value.
+   */
+  readonly progressValueFormatter?: (
+    currentValue: number,
+    minValue?: number,
+    maxValue?: number
+  ) => string
+  /**
+   * An icon that can provide visual information
+   */
+  readonly icon?: Readonly<JSX.Element>
+}
+
+/**
+ * Primary UI component for progress of a treatment.
+ */
+export const ProgressBar: React.FC<TProgressBarProps> = ({
+  label,
+  minValue = 0,
+  maxValue = 100,
+  currentValue,
+  currentValueFormatter = (currentValue: number): string => `${currentValue}`,
+  progressValueFormatter = (currentValue: number, minValue?: number, maxValue?: number): string =>
```

Comment:
I think it would be nice to make these functions exported knowing that they can be re-used. I mean:

```typescript
export const defaultValueFormatter = (currentValue: number): string => `${currentValue}`
export const defaultProgressValueFormatter = (currentValue: number, minValue?: number, maxValue?: number): string => `${toPercent(currentValue, minValue ?? 0, maxValue ?? 100).toFixed(2)} %`
```

Reply:
Ok that's a good idea